tests/generation/SettingsTest.py

```diff
@@ -42,6 +50,32 @@ def test_write_info(self):
 
         self.assertEqual(result_comment, line)
 
+    def test_allow_complete_section_mode_with_ignore_globs(self):
+        project_dir = "/repo"
```

Comment:
@ishanSrt Here we aren't testing the path collection (there are separate tests for that I guess), we've assumed the paths and we are testing the functions which take in these paths and do some stuff (generate settings in this case). It's like instead of starting from highest level function (entry point) we're starting from somewhere in between. This actually saves time while testing, and that's what you can call as "unit testing" (testing specific things one at a time).